gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -63,10 +63,8 @@
 @Slf4j
 @Getter
 public class IcebergDataset implements PrioritizedCopyableDataset {
-  private final String dbName;
-  private final String inputTableName;
   private final IcebergTable srcIcebergTable;
-  /** Presumed destination {@link IcebergTable} exists */
+  /* CAUTION: *hopefully* `destIcebergTable` exists... although that's not necessarily been verified yet */
```

Comment:
Might be out of scope for this PR, but shouldn't our default behavior not have that assumption and create the table if missing? That's the contract for Hive distcp

Reply:
for some catalogs, like openhouse, that's not actually possible: the tables must have been already mated in a way that we are unable to accomplish ourselves (outside of the catalog itself)